src/services/refactors/addAsyncSuffix.ts

```diff
@@ -0,0 +1,51 @@
+/* @internal */
+namespace ts.refactor {
+
+    const asyncSuffixRefactor: Refactor = {
+        name: "Add Async suffix",
+        description: "Add an 'Async' suffix to async function declarations",
+        getCodeActions,
+        isApplicableForPositionOrRange
+    };
+
+    registerRefactor(asyncSuffixRefactor);
+
+    function getCodeActions(context: RefactorContext, positionOrRange: number | TextRange): CodeAction[] | undefined {
+        const { boundSourceFile, program } = context;
+        const tokenPos = typeof positionOrRange === "number" ? positionOrRange : positionOrRange.pos;
+        const token = getTokenAtPosition(boundSourceFile, tokenPos);
+
+        const functionSymbol = program.getTypeChecker().getSymbolAtLocation(token);
+        if (!(functionSymbol.flags & SymbolFlags.Function)) {
+            return undefined;
+        }
+
+        const oldNameNode = functionSymbol.valueDeclaration.name as Identifier;
+        const changeTracker = textChanges.ChangeTracker.fromCodeFixContext(context);
+
+        changeTracker.replaceNode(boundSourceFile, oldNameNode, createIdentifier(functionSymbol.name + "Async"));
+
+        return [{
+            changes: changeTracker.getChanges(),
+            description: asyncSuffixRefactor.description
+        }];
+    }
+
+    function isApplicableForPositionOrRange(context: LightRefactorContext, positionOrRange: number | TextRange): boolean {
+        const { nonBoundSourceFile } = context;
+        const tokenPos = typeof positionOrRange === "number" ? positionOrRange : positionOrRange.pos;
+        const token = getTokenAtPosition(nonBoundSourceFile, tokenPos);
+
+        let node = token;
+        while (node) {

```

Comment:
It is kind of random that on  any expression i can refactor the containing function to include `async`. i would say this has to be on function/method declaration at least.